client/utils/validation.js

```diff
@@ -1,35 +1,56 @@
+import {
+  confirmed,
+  email,
+  is_not as isNot,
+  max,
+  max_value as maxValue,
+  min,
+  min_value as minValue,
+  numeric,
+  required
+} from 'vee-validate/dist/rules';
+import { extend } from 'vee-validate';
+import forEach from 'lodash/forEach';
+import snakeCase from 'lodash/snakeCase';
 import userApi from '@/api/user';
-import VeeValidate from 'vee-validate';
+
+const URL_REGEX = new RegExp(
+  `(https?://(?:www.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9].[^s]{2,}
+  |www.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9].[^s]{2,}|https?://(?:www.|(?!www))
+  [a-zA-Z0-9]+.[^s]{2,}|www.[a-zA-Z0-9]+.[^s]{2,})$`);
 
 const alphanumerical = {
-  getMessage: field => {
-    return `The ${field} field must contain at least 1 letter and 1 numeric value.`;
-  },
-  validate: value => {
-    return (/\d/.test(value) && /[a-zA-Z]/.test(value));
-  }
+  validate: value => (/\d/.test(value) && /[a-zA-Z]/.test(value)),
+  message: 'The {_field_} field must contain at least 1 letter and 1 numeric value.'
 };
 
 const uniqueEmail = {
-  getMessage: field => `The ${field} is not unique.`,
-  validate: (email, user) => {
-    if (user && email === user.email) return true;
-    return userApi.fetch({ email }).then(({ total }) => ({ valid: !total }));
-  }
+  params: ['userData'],
+  validate: (email, { userData }) => {
+    if (userData && email === userData.email) return true;
+    return userApi.fetch({ email }).then(({ total }) => !total);
+  },
+  message: 'The {_field_} is not unique.'
 };
 
-VeeValidate.Validator.extend('alphanumerical', alphanumerical);
-VeeValidate.Validator.extend('unique-email', uniqueEmail);
-
-export default VeeValidate;
+const url = {
+  validate: value => URL_REGEX.test(value),
+  message: 'The {_field_} is invalid.'
+};
 
-const mixin = ({ inherit = false } = {}) => {
-  if (inherit) return { inject: ['$validator'] };
-  return {
-    $_veeValidate: {
-      validator: 'new'
-    }
-  };
+const rules = {
+  alphanumerical,
+  confirmed,
+  email,
+  isNot,
+  max,
+  maxValue,
+  min,
+  minValue,
+  numeric,
+  required,
+  uniqueEmail,
+  url
 };
 
-export const withValidation = mixin;
+forEach(rules, (rule, name) => extend(snakeCase(name), rule));
```

Comment:
This will not work for non-custom rules and the ones that do not have message property set. Instead, `FIELD_NAME is not valid.` message will be displayed. 
To fix this:
- import `messages`:
```
import { messages } from 'vee-validate/dist/locale/en.json';
``` 
- rewrite this line to:
```
forEach(rules, (rule, name) => extend(snakeCase(name), { message: messages[name], ...rule }));
```